### PR TITLE
[DOC-3559] Build delegate with immutable image type tutorial

### PR DESCRIPTION
**Pull request by brian-f-harness**
Build delegate with immutable image type tutorial

For reviewers, a preview is available.

[Configure a delegate with the minimal image type](https://64d565c1ca4ca40260408a01--harness-developer.netlify.app/tutorials/platform/secure-delegate-default-to-minimal)

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — APPROVED** [review]

**Review by thisrohangupta — APPROVED** [review]

File: tutorials/platform/secure-delegate-default-to-minimal.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
+---
+title: Build and set up a delegate with an immutable image type
+description: This tutorial shows you how to build and set up a delegate with an immutable image type.
+sidebar_label: Move from the delegate minimal image
+sidebar_position: 6
+keywords: [delegate,delegate image]
+---
+
+Harness recommends that you use the Harness Delegate minimal image (*`yy.mm.xxxxx.minimal`*) when you set up Harness Platform for production use. This image has been thoroughly scanned and is free of any high or critical vulnerabilities. Users focused on security tend to prefer this option. However, the minimal delegate image lacks some default binaries, which require you to configure your delegates and install necessary binaries to ensure that Continuous Deployment (CD) steps function properly and remain vulnerability-free from third-party tools. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```
Comment (michaelcretzman) [inline tutorials/platform/secure-delegate-default-to-minimal.md:9]:
```suggestion
Harness recommends that you use the Harness Delegate minimal image (*`yy.mm.xxxxx.minimal`*) when you set up the Harness Platform for production use. This image has been thoroughly scanned and is free of any high or critical vulnerabilities. Users focused on security tend to prefer this option. 

However, the minimal delegate image lacks some binaries that are required for Continuous Deployment (CD) steps to function properly and remain vulnerability-free from third-party tools. Consequently, using the minimal delegate image requires you to configure your delegates and install necessary binaries. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

File: tutorials/platform/secure-delegate-default-to-minimal.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+The Harness Delegate minimal image (*`yy.mm.xxxxx.minimal`*) is a lighter, more secure version of the default Harness Delegate image. Its main purpose is to provide an enhanced security profile for users, especially those who prioritize their systems' security. The Harness Delegate minimal images includes the following features.
+
+- **Security Scanned:** The image undergoes rigorous scanning processes to ensure that it is devoid of any high-risk or critical vulnerabilities. This makes it an optimal choice for organizations or users who have stringent security requirements.
+
+- **Limited** Binaries: Unlike the standard delegate, the minimal image does not come packed with all the default binaries. While this contributes to its lightweight nature and security, it also means that users have additional responsibilities. They must manually configure and add any necessary binaries to make their setup functional.
```
Comment (michaelcretzman) [inline tutorials/platform/secure-delegate-default-to-minimal.md:15]:
```suggestion
- **Limited Binaries:** Unlike the standard delegate, the minimal image does not include all of the default binaries. While this contributes to its lightweight nature and security, it also means that users have additional responsibilities. They must manually configure and add any necessary binaries to make their setup functional.
```

File: tutorials/platform/secure-delegate-default-to-minimal.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+- **User Responsibilities:** Because the minimal delegate image is devoid of the default binaries, users are in charge of tailoring it to their needs. This includes installing specific binaries essential for their CD steps. This level of control also allows users to maintain an updated environment. By installing the latest versions of necessary binaries, they can ensure that the delegate remains free from potential vulnerabilities found in outdated third-party tools.
+
+- **Preferred by Security-Conscious Users:** Due to its clean security slate, many users who prioritize system security gravitate towards the minimal delegate image. By starting with a minimal setup and adding only what is necessary, they can maintain a tighter control over the software and tools present, thus minimizing potential security risks.
+
+This tutorial shows you how to build and set up a delegate with  an immutable image type (*`yy.mm.xxxxx.minimal`*) for CD.
```
Comment (michaelcretzman) [inline tutorials/platform/secure-delegate-default-to-minimal.md:21]:
```suggestion
This tutorial shows you how to build and set up a minimal delegate image (*`yy.mm.xxxxx.minimal`*) for Harness CD.
```

File: tutorials/platform/secure-delegate-default-to-minimal.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+3. Build and push
+   
+   ```
+   docker buildx build --platform linux/amd64 --output type=docker .
+   docker image ls
+   docker tag 34594c2f0882 harness/delegate:yy.mm.xxxxx.harness.custom.x86.v4
+   docker push harness/delegate:yy.mm.xxxxx.harness.custom.x86.v4
+   ```
+
+4. Update your delegate image in the manifest with the tag from the image you pushed in previous step.
```
Comment (michaelcretzman) [inline tutorials/platform/secure-delegate-default-to-minimal.md:36]:
```suggestion
4. In the delegate image manifest, update the delegate image with the tag from the image you pushed in the previous step.
```

Comment (michaelcretzman) [reply tutorials/platform/secure-delegate-default-to-minimal.md:36]: this needs an example, imo. I don't see a manifest with the tag updated, or even a snippet. So how does the user know where to do it?

**Review by michaelcretzman — DISMISSED** [review]
Approved with a few suggestions.

File: tutorials/platform/secure-delegate-default-to-minimal.md (hunk 4 above)
Comment (brian-f-harness) [reply tutorials/platform/secure-delegate-default-to-minimal.md]: @michaelcretzman nice catch. Thank you. Worked with @pratmit to add an xref here.

**Review by pratmit — APPROVED** [review]

File: tutorials/platform/secure-delegate-default-to-minimal.md [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+ namespace: harness-delegate-ng  
+spec:  
+ type: ClusterIP  
+ selector:  
+   harness.io/name: custom-del  
+ ports:  
+   - port: 8080  
+   
+---  
+
```
Comment (ivbalan-harness) [inline tutorials/platform/secure-delegate-default-to-minimal.md:244]: Everything below this line is not needed if upgrader job is removed (original yaml had it commented out)

**Pull request merged**